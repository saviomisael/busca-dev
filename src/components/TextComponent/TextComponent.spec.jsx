import { screen } from '@testing-library/react';
import { TextComponent } from '.';

describe('<TextComponent />', () => {
  it('should renders component with default styles', () => {
    global.renderTheme(<TextComponent>Children</TextComponent>);

    const text = screen.getByText('Children');

    expect(text).toHaveStyleRule('font-size', '1.8rem');
  });

  it('should renders component with smal font size', () => {
    global.renderTheme(<TextComponent smallText>Children</TextComponent>);

    const text = screen.getByText('Children');

    expect(text).toHaveStyleRule('font-size', '1.6rem');
  });

  it('should matches with snapshot', () => {
    global.renderTheme(<TextComponent>Children</TextComponent>);

    const text = screen.getByText('Children');

    expect(text).toMatchSnapshot();
  });
});
