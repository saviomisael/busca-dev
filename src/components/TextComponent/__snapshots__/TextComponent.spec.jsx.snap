// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextComponent /> should matches with snapshot 1`] = `
.c0 {
  font-size: 1.8rem;
  color: #F5F5F5;
}

<p
  class="c0"
>
  Children
</p>
`;
